spec/lib/test_applications_spec.rb

```diff
@@ -105,4 +105,37 @@
       expect(choices.first.application_form.application_work_history_breaks.count).to eq(1)
     end
   end
+
+  describe 'reference completion' do
+    it 'completes all references for unsubmitted_with_completed_references applications' do
+      courses_we_want = create_list(:course_option, 2, course: create(:course, :open_on_apply)).map(&:course)
+
+      application_choice = TestApplications.new.create_application(recruitment_cycle_year: 2021, states: %i[unsubmitted_with_completed_references], courses_to_apply_to: courses_we_want).first
+
+      references = application_choice.application_form.application_references
+
+      expect(references.any?(&:feedback_requested?)).to be false
```

Comment:
Maybe check that all the references are `feedback_provided` rather than not `feedback_requested`? Just in case they've got into a weird intermediate state